packages/core/src/api/app/FeatureFlags.tsx

```diff
@@ -0,0 +1,222 @@
+class FeatureFlagsImpl implements FeatureFlagsApi {
+  private readonly localStorageKey = 'featureFlags';
+
+  private get(
+    enabledFeatureFlags: Set<string>,
+    name: FeatureFlagName,
+  ): FeatureFlagState {
+    if (enabledFeatureFlags.has(name)) {
+      return FeatureFlagState.Enabled;
+    }
+
+    return FeatureFlagState.NotEnabled;
+  }
+
+  private set(name: FeatureFlagName, state: FeatureFlagState): void {
+    const errors = this.checkFeatureFlagNameErrors(name);
+    const flags = this.getEnabledFeatureFlags();
+
+    if (errors.length > 0) {
+      throw new Error(errors[0]);
+    }
+
+    if (state === FeatureFlagState.Enabled) {
+      flags.add(name);
+    } else if (state === FeatureFlagState.NotEnabled) {
+      flags.delete(name);
+    } else {
+      throw new Error(
+        'The `state` argument requires a recognized value from the FeatureFlagState enum. ' +
+          'Please check the Backstage documentation to see all the available options.' +
+          'Example values: FeatureFlagState.NotEnabled, FeatureFlagState.Enabled',
+      );
+    }
+
+    window.localStorage.setItem(
+      this.localStorageKey,
+      JSON.stringify(
+        [...flags].reduce((list, flag) => ({ ...list, [flag]: true }), {}),
+      ),
+    );
+  }
+
+  getEnabledFeatureFlags(): Set<FeatureFlagName> {
+    if (!('localStorage' in window)) {
+      throw new Error(
+        'Feature Flags are not supported on browsers without the Local Storage API',
+      );
+    }
+
+    try {
+      const featureFlagsJson = window.localStorage.getItem(
+        this.localStorageKey,
+      );
+      return new Set<FeatureFlagName>(
+        Object.keys(JSON.parse(featureFlagsJson!)),
+      );
+    } catch (err) {
+      return new Set<FeatureFlagName>();
+    }
+  }
+
+  // We don't make this private as we need this to validate
+  // in the `registerFeatureFlag` method in the Plugin API.
+  checkFeatureFlagNameErrors(name: FeatureFlagName): string[] {
+    const errors: string[] = [];
+
+    if (name.length < 3) {
+      errors.push('The name must have a minimum length of three characters.');
+    }
+
+    if (name.length > 150) {
+      errors.push('The name must not exceed 150 characters.');
+    }
+
+    if (!name.match(/^[a-z]+[a-z0-9-]+$/)) {
+      errors.push(
+        'The name must start with a lowercase letter and only contain lowercase letters, numbers and hyphens. ' +
+          'Examples: feature-flag-one, alpha, release-2020',
+      );
+    }
+
+    return errors;
+  }
+
+  useFeatureFlag(
+    name: FeatureFlagName,
+  ): [FeatureFlagState, (state: FeatureFlagState) => void] {
+    // Check for context
+    const context = useContext(FeatureFlagsContext);
+    if (!context) {
+      throw new Error(
+        'No FeatureFlagsContext found. ' +
+          'Please use this React Hook in the context of your <App />',
+      );
+    }
+
+    // Check for errors
+    // eslint-disable-next-line no-use-before-define
```

Comment:
Sup with all of these? use `this` :grin:

Reply:
`this` wasn't working and only got `undefined` and couldn't seem to figure ut what the issue is. I'm only using it twice I think anyways

Reply:
Ah right, it's because of the destructuring when consuming the hook, would've had to define this with an arrow function instead of the method syntax.

Reply:
Fixed. No longer using this.